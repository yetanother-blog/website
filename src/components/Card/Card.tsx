import React from 'react';
import styled, { useTheme } from 'styled-components';
import { Link as GatsbyLink } from 'gatsby';
import {
  width,
  margin,
  padding,
  borderRadius,
  MarginProps,
  WidthProps,
  BorderRadiusProps,
  PaddingProps,
} from 'styled-system';

interface Props {
  to: string;
}

export type CardProps = Props & MarginProps & WidthProps;

export type StyledSystemProps = Props & BorderRadiusProps & PaddingProps & Omit<CardProps, 'to'>;

export const StyledCard = styled(GatsbyLink)<StyledSystemProps>`
    display: block;
    background-color: ${(props) => props.theme.colors.grey200};
    text-decoration: none;
    border-radius: ${(props) => props.theme.radii.s};
    color: inherit;
    font-family: inherit;
    transition: background-color 200ms ease-in;
    box-sizing: border-box;

    ${padding}
    ${width}
    ${margin}
    ${borderRadius}

    &:hover,
    &:focus {
      background-color: ${(props) => props.theme.colors.grey300};
    }
`;

export const Card: React.FC<CardProps> = ({ children, to, ...styledProps }) => {
  const theme = useTheme();

  return (
    <StyledCard p={[theme.space.l, theme.space.xl]} to={to} {...styledProps}>
      {children}
    </StyledCard>
  );
};
