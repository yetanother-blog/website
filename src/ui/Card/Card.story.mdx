import { Story, Meta, Preview, Props } from '@storybook/addon-docs/blocks';
import { Card } from './Card';
import { ThemeContext } from 'styled-components';
import { Typography } from '../Typography/Typography';
import { Link } from '../Link/Link';
import { Box } from '../Box/Box';

<Meta title="Components|Card" />

# Card

Card component to seperate blog formats.

<Preview>
  <Story name="default">
    <ThemeContext.Consumer>
      {(theme) => (
        <Box
          display="flex"
          flexDirection={['column', 'row']}
          justifyContent="space-between"
          marginBottom={theme.space.m}
        >
          <Card width={['100%', 'calc(50% - 10px)']} to="/" marginBottom={theme.space.l}>
            <Link display="block" variant="primary" component="span" marginBottom={theme.space.m}>
              repo of the week
            </Link>
            <Typography variant="smallText">
              Check out the latest indipendent repository on the internet. We provide you opinionated information.
            </Typography>
          </Card>
          <Card width={['100%', 'calc(50% - 10px)']} to="/" marginBottom={theme.space.l}>
            <Link display="block" variant="secondary" component="span" marginBottom={theme.space.m}>
              guide of the month
            </Link>
            <Typography variant="smallText">
              On a monthly basis we create valueble guides about the latest technologies on the web.
            </Typography>
          </Card>
        </Box>
      )}
    </ThemeContext.Consumer>
  </Story>
</Preview>

## Props

<Props of={Card} />
