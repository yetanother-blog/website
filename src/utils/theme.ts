import { DefaultTheme } from 'styled-components';

export const theme: DefaultTheme = {
  fontSizes: ['12px', '14px', '16px', '20px', '24px', '32px'],
  colors: {
    primary: '#00FF00',
    secondary: '#FB00FF',
    grey500: '#1C1C1C',
    grey400: '#ACADAC',
    grey300: '#F6F6F6',
  },
  breakpoints: ['576px', '768px', '992px', '1200px'],
  space: {
    zero: 0,
    xxs: 4,
    xs: 8,
    s: 12,
    m: 16,
    l: 20,
    xl: 40,
    xxl: 60,
    xxxl: 120,
  },
  radii: {
    s: 3,
  },
  width: {
    m: '620px',
    l: '940px',
    xl: '1200px',
  },
};
