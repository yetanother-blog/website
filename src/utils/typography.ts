import { CSSObject } from "styled-components";
import Typography from "typography";

const typography = new Typography({
  googleFonts: [
    {
      name: "Source Sans Pro",
      styles: ["400", "600"]
    },
    {
      name: "Source Code Pro",
      styles: ["600", "700"]
    }
  ],
  baseFontSize: "20px",
  baseLineHeight: 1.85,
  bodyFontFamily: ["Source Sans Pro", "sans-serif"],
  boldWeight: "600",
  overrideStyles: ({ adjustFontSizeTo }, options, styles) => ({
    a: {
      fontFamily: `"Source Code Pro", "sans-serif"`
    }
  })
});

// Hot reload typography in development.
if (process.env.NODE_ENV !== `production`) {
  typography.injectStyles();
}

export default typography;

export const rhythm = typography.rhythm;
export const scale = typography.scale;

type StyledScale = (values: number) => CSSObject;
export const styledScale = scale as StyledScale;
