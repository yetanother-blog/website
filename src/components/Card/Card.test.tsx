import React from 'react';
import { render } from '@testing-library/react';
import { ThemeContext } from 'styled-components';
import { theme } from '../../utils/theme';
import { Card } from './Card';

describe('SocialLink component', () => {
  it('should render social link for twitter', () => {
    const { container } = render(
      <ThemeContext.Provider value={theme}>
        <Card to="/">Some Content</Card>
      </ThemeContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
