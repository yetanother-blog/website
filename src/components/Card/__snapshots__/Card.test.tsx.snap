// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLink component should render social link for twitter 1`] = `
<div>
  <a
    class="sc-bdvvaa cnDazs"
    href="/"
    p="20,40"
  >
    Some Content
  </a>
</div>
`;
